src/editors/Editor.test.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { Editor, mapDispatchToProps } from './Editor';
+import { thunkActions } from './data/redux';
+import * as hooks from './hooks';
+import { blockTypes } from './data/constants/app';
+
+jest.mock('./hooks', () => ({
+  initializeApp: jest.fn(),
+  prepareEditorRef: jest.fn().mockName('prepareEditorRef'),
+}));
+
+jest.mock('./containers/TextEditor/TextEditor', () => 'TextEditor');
+jest.mock('./containers/VideoEditor/VideoEditor', () => 'VideoEditor');
+jest.mock('./containers/ProblemEditor/ProblemEditor', () => 'ProblemEditor');
+jest.mock('./components/EditorFooter', () => 'EditorFooter');
+jest.mock('./components/EditorHeader', () => 'EditorHeader');
+
+const props = {
+  courseId: 'course-v1:edX+DemoX+Demo_Course',
+  blockId: 'block-v1:edX+DemoX+Demo_Course+type@html+block@030e35c4756a4ddc8d40b95fbbfff4d4',
+  studioEndpointUrl: 'fakeurl.com',
+  initialize: jest.fn(),
+};
+
+describe('Editor', () => {
+  describe('snapshots', () => {
+    test('renders no editor when ref isnt ready', () => {
+      hooks.prepareEditorRef.mockImplementationOnce(
+        () => ({ editorRef: null, refReady: false, setEditorRef: jest.fn() }),
+      );
+      expect(shallow(<Editor blockType="AblOck" {...props} />)).toMatchSnapshot();
+    });
+    test.each(Object.values(blockTypes))('renders %p editor when ref is ready', (blockType) => {
```

Comment:
I hate to say this, but I think this should be split out into separate tests, explicitly spelling out WHICH component is expected to be rendered for each blockType.
What might make sense here, is to do a single snapshot test for a supported type, and then for each type, have a simple test that renders the component and checks the type of the editor component.  The snapshots changing only one field out the larger component are repetitive and harder to verify.

Reply:
So here, remove the per-blockType snapshot, and add a single test for 1 block type that takes a snapshot, and verifies that the editor spawns once and only once, in the right place.